### PR TITLE
mgr/dashboard: Add namespaces views in dashboard     

### PR DESCRIPTION
**Pull request by afreen23**
Fixes https://tracker.ceph.com/issues/66990

- list namespaces
- create namespaces
- edit namespaces
- delete namespaces
- update NQN regex

[Screencast from 2024-07-17 08-17-48.webm](https://github.com/user-attachments/assets/c00e5ef7-1b15-4f76-ae52-b50301ef9609)

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by afreen23** [comment]
jenkins test api

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-namespaces-form/nvmeof-namespaces-form.component.html [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                     name="image_size"
+                     formControlName="image_size">
+              <select id="unit"
+                      name="unit"
+                      class="form-input form-select"
+                      formControlName="unit">
+                <option *ngFor="let u of units"
+                        [value]="u"
+                        i18n>{{ u }}</option>
+              </select>
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-namespaces-form/nvmeof-namespaces-form.component.html:92]: probably we should look into replacing our other Size fields (in rbd images, cephfs etc) to this format. `textbox` and `units`. Can you open a tracker to do that?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-namespaces-form/nvmeof-namespaces-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    if (this.router.url.includes('subsystems/(modal:edit')) {
+      this.initForEdit();
+    } else {
+      this.initForCreate();
+    }
+  }
+
+  createForm() {
+    this.nsForm = new CdFormGroup({
+      image: new UntypedFormControl(`image:${Date.now()}`, {
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-namespaces-form/nvmeof-namespaces-form.component.ts:108]: to better distinguish it?
```suggestion
      image: new UntypedFormControl(`nvme_ns_image:${Date.now()}`, {
```

**Review by nizamial09 — COMMENTED** [review]
overall lgtm! thank you @afreen23

**Comment by afreen23** [comment]
Updates summary:
* addressed review comments

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-namespaces-form/nvmeof-namespaces-form.component.html (hunk 1 above)
Comment (afreen23) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/nvmeof-namespaces-form/nvmeof-namespaces-form.component.html:92]: https://tracker.ceph.com/issues/66995

**Comment by nizamial09** [comment]
jenkins test make check

**Comment by nizamial09** [comment]
jenkins test api

**Comment by afreen23** [comment]
jenkins test api

**Comment by afreen23** [comment]
jenkins test make check

**Comment by afreen23** [comment]
jenkins test api

**Comment by afreen23** [comment]
jenkins test api

**Comment by afreen23** [comment]
jenkins test dashboard

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by afreen23** [comment]
jenkins test make check arm64

**Comment by afreen23** [comment]
jenkins test make check

**Comment by afreen23** [comment]
jenkins test api

**Comment by afreen23** [comment]
jenkins test make check

**Comment by afreen23** [comment]
jenkins test dashboard

**Comment by afreen23** [comment]
jenkins test dashboard cephadm

**Comment by afreen23** [comment]
jenkins test make check arm64

**Comment by afreen23** [comment]
jenkins test api

**Comment by afreen23** [comment]
jenkins test make check

**Comment by afreen23** [comment]
jenkins test api

**Comment by afreen23** [comment]
jenkins test make check

**Comment by afreen23** [comment]
jenkins test make check

**Comment by afreen23** [comment]
jenkins test api

**Comment by cloudbehl** [comment]
```

**Comment by afreen23** [comment]
Updates summary:

* rebase
* fixing of tests

**Comment by afreen23** [comment]
Updates summary:

* rebase
* fixing of tests

**Comment by cloudbehl** [comment]
jenkins test api

**Comment by cloudbehl** [comment]
jenkins test make check

**Comment by afreen23** [comment]
Updates summary:
* updated unit tests , failing die to name changes after taking in the reviews

**Comment by afreen23** [comment]
Dashboard tests unrelated:

[Test Result](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/15727/testReport/) (4 failures / ±0)

    [Create a CephFS Volume.CephFS Subvolume Group management Create a CephFS Volume](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/15727/testReport/junit/(root)/Create%20a%20CephFS%20Volume/CephFS_Subvolume_Group_management_Create_a_CephFS_Volume/)
    [Create a CephFS Volume.CephFS Subvolume management Create a CephFS Volume](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/15727/testReport/junit/(root)/Create%20a%20CephFS%20Volume/CephFS_Subvolume_management_Create_a_CephFS_Volume/)
    [Create a CephFS Volume.CephFS Management Create a CephFS Volume](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/15727/testReport/junit/(root)/Create%20a%20CephFS%20Volume/CephFS_Management_Create_a_CephFS_Volume/)
    [Create a CephFS Volume.CephFS Snapshot Management Create a CephFS Volume](https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/15727/testReport/junit/(root)/Create%20a%20CephFS%20Volume/CephFS_Snapshot_Management_Create_a_CephFS_Volume/)

**Comment by afreen23** [comment]
jenkins test make check

**Comment by rivkap** [comment]
@afreen23 the helper text in pool should be "An RBD application-enabled pool..." not "A RBD application-enabled pool...." (AN not A). Thanks

**Pull request merged**